web/components/mod-report-item.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react'
+import { UserHovercard } from 'web/components/user/user-hovercard'
+import { Avatar } from 'web/components/widgets/avatar'
+import { UserLink } from 'web/components/widgets/user-link'
+import { RelativeTimestamp } from 'web/components/relative-timestamp'
+import { ChoicesToggleGroup } from 'web/components/widgets/choices-toggle-group'
+import { EditableModNote } from 'web/components/contract/editable-mod-note'
+import { BannedBadge } from 'web/components/widgets/user-link'
+import Link from 'next/link'
+import { ModReport, ReportStatus } from 'common/mod-report'
+import { Col } from './layout/col'
+import { Row } from './layout/row'
+
+interface ReportItemProps {
+  report: ModReport
+  reportStatuses: { [key: number]: ReportStatus }
+  modNotes: { [key: number]: string | undefined }
+  handleStatusChange: (
+    reportId: number,
+    newStatus: ReportStatus
+  ) => Promise<void>
+  handleNoteSave: (reportId: number, newNote: string) => Promise<void>
+}
+
+const ModReportItem: React.FC<ReportItemProps> = ({
+  report,
+  reportStatuses,
+  modNotes,
+  handleStatusChange,
+  handleNoteSave,
+}) => {
+  const extractTextFromJSONContent = (content: any): string => {
+    if (!content) {
+      console.warn('No content provided')
+      return ''
+    }
+
+    if (!Array.isArray(content)) {
+      console.warn('Content is not an array', content)
+      return ''
+    }
+
+    let text = ''
+    content.forEach((node: any) => {
+      if (node.type === 'text') {
+        text += node.text
+      } else if (node.type === 'mention' && node.attrs) {
+        text += `@${node.attrs.label} `
+      } else if (node.type === 'paragraph' && node.content) {
+        text += extractTextFromJSONContent(node.content) + '\n'
+      } else if (node.content) {
+        text += extractTextFromJSONContent(node.content)
+      } else {
+        console.warn('Unknown node type or missing content', node)
+      }
+    })
+    return text.trim()
+  }
+
+  const parseCommentContent = (commentContent: any) => {
+    if (typeof commentContent === 'string') {
+      try {
+        return JSON.parse(commentContent)
+      } catch (error) {
+        console.error('Failed to parse comment content', error)
+        return null
```

Comment:
This is definitely not the right way to do this. I would try to use the rich text library to parse the content. Maybe you can ask @sinclairity?

I would be ok with you fixing this in a follow-up PR though if you add a TODO.

Reply:
`{richTextToString(comment.content)}`